storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteStorageThreadPool.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.storage.internals.log;
+
+import org.apache.kafka.common.internals.FatalExitError;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class RemoteStorageThreadPool extends ThreadPoolExecutor {
+    private final Logger logger;
+
+    public RemoteStorageThreadPool(String threadNamePrefix,
+                                   int numThreads,
+                                   int maxPendingTasks) {
+        super(numThreads, numThreads, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(maxPendingTasks),
+                new RemoteStorageThreadFactory(threadNamePrefix));
+        logger = new LogContext() {
+            @Override
+            public String logPrefix() {
```

Comment:
@satishd pardon me, why we don't use the constructor to set prefix? the method `logger` does not use the value from override method, and hence the prefix is never propagated to logger.

I have filed https://issues.apache.org/jira/browse/KAFKA-17168 for it, and please feel free to correct me. Maybe I misunderstand this override method

Reply:
@chia7712 I think it was added as part of `logPrefix()` method implementation but not as part of the constructor because the logger should have the invoking thread name as the prefix but not the thread name that created this thread pool.

Reply:
why do we have to prefix the thread name in the logger? The user can include (%t) in conversion pattern to print the thread name in the logger.

Reply:
> because the logger should have the invoking thread name as the prefix but not the thread name that created this thread pool.

That is a good reason ( if they don't want to use `ConversionPattern` in log4j properties). However, I'm not sure whether overriding `logPrefix` can address that since we create `Logger` by the inner variable rather the override method.

It seems the follow-up should use `LogContext()`. As @kamalcph mentioned, using `ConversionPattern` should be good enough.

Reply:
Good points @chia7712  and @kamalcph. I agree users can always set a conversion pattern that includes thread-name, but I do not remember any other reason why this prefix was added. :) 

I suggest completely removing logPrefix from the new code instead of using the name of the thread that created threadpool which is not useful anyway.

Reply:
Opened #16657 to address this comment.